x-pack/plugin/logsdb/qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/LogsIndexModeRollingUpgradeIT.java

```diff
@@ -103,75 +115,34 @@ protected Settings restClientSettings() {
+
         } else if (isMixedCluster()) {
-            assertOK(client().performRequest(rolloverDataStream(client(), "logs-apache-production")));
-            final Response bulkIndexResponse = client().performRequest(bulkIndex("logs-apache-production", () -> {
-                final StringBuilder sb = new StringBuilder();
-                for (int i = 0; i < randomIntBetween(10, 20); i++) {
-                    sb.append(
-                        String.format(
-                            BULK_INDEX_REQUEST,
-                            DateFormatter.forPattern(FormatNames.DATE_TIME.getName()).format(Instant.now()),
-                            randomFrom("foo", "bar"),
-                            randomFrom("PUT", "POST", "GET"),
-                            InetAddresses.toAddrString(randomIp(randomBoolean())),
-                            randomIntBetween(20, 50)
-                        )
-                    );
-                    sb.append("\n");
-                }
-                return sb.toString();
-            }));
-            assertOK(bulkIndexResponse);
-            assertThat(entityAsMap(bulkIndexResponse).get("errors"), Matchers.is(false));
+            // when/then - index more documents
+            bulkIndex(this::bulkIndexRequestBody);
+
         } else if (isUpgradedCluster()) {
+            // when/then - index some more documents
+            bulkIndex(this::bulkIndexRequestBody);
+
+            // given - enable logsdb and rollover
             enableLogsdbByDefault();
-            assertOK(client().performRequest(rolloverDataStream(client(), "logs-apache-production")));
-            final Response bulkIndexResponse = client().performRequest(bulkIndex("logs-apache-production", () -> {
-                final StringBuilder sb = new StringBuilder();
-                for (int i = 0; i < randomIntBetween(10, 20); i++) {
-                    sb.append(
-                        String.format(
-                            BULK_INDEX_REQUEST,
-                            DateFormatter.forPattern(FormatNames.DATE_TIME.getName()).format(Instant.now()),
-                            randomFrom("foo", "bar"),
-                            randomFrom("PUT", "POST", "GET"),
-                            InetAddresses.toAddrString(randomIp(randomBoolean())),
-                            randomIntBetween(20, 50)
-                        )
-                    );
-                    sb.append("\n");
-                }
-                return sb.toString();
-            }));
-            assertOK(bulkIndexResponse);
-            assertThat(entityAsMap(bulkIndexResponse).get("errors"), Matchers.is(false));
+            rolloverDataStream();
+
+            // when/then
+            bulkIndex(this::bulkIndexRequestBody);
 
+            // then continued - verify that only the latest write index has logsdb enabled
             assertIndexSettings(0, Matchers.nullValue());
-            assertIndexSettings(1, Matchers.nullValue());
```

Comment:
since we no longer rollover when the cluster is in a mixed state, we have only two indices at the end